cache/fifo_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package cache
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFIFOCache(t *testing.T) {
+	tests := []struct {
+		name     string
+		limit    int
+		maxVal   int
+		expected error
+		check    func(t *testing.T, cache *FIFO[int, int])
+	}{
+		{
+			name:     "empty cache fails",
+			limit:    0,
+			expected: errors.New("maxSize must be greater than 0"),
+		},
+		{
+			name:   "inserting up to limit works",
+			limit:  10,
+			maxVal: 9,
+		},
+		{
+			name:   "inserting after limit cleans first",
+			limit:  10,
+			maxVal: 10,
+			check: func(t *testing.T, cache *FIFO[int, int]) {
+				require := require.New(t)
+				_, ok := cache.Get(0)
+				require.False(ok)
+				v, ok := cache.Get(1)
+				require.True(ok)
+				require.Equal(1, v)
+			},
+		},
+		{
+			name:   "inserting before limit can insert again",
+			limit:  10,
+			maxVal: 8,
+			check: func(t *testing.T, cache *FIFO[int, int]) {
+				require := require.New(t)
+				exists := cache.Put(9, 9)
```

Comment:
Usually it's idiomatic to use `ok` in Go instead of `exists`

Reply:
Mhh.. `ok` seems to suggest that something can go wrong, like inserting twice in a database that would not accept that. Here it's returning `true` if the value was already in the cache but that doesn't mean that it's not going to insert the changes afterall

Reply:
For `Get` it makes more sense though!

Reply:
I think that we should only use `ok` instead of `exists` for `Get` because `ok` seems to imply that something can go wrong while it just means that the value is already in the cache so no need to insert it again. It seems like `ok` could be used if the cache does not support overwriting value for instance.